src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/NavigationBar.tsx

```diff
@@ -0,0 +1,217 @@
+      display: 'none'
+    }
+  },
+  li2Mounted: {
+    marginBottom: '6px',
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  li3: {
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  a: {
+    display: 'block', 
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    }
+  },
+  a2: {
+    display: 'block',   
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    color: 'white!important',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    },
+    width: '100%',
+    fontSize: '1.5rem'
+  },
+  i: {
+    color: 'white'
+  },
+  hide: {
+    display: 'none'
+  }
+});
+
+export interface INavigationBar {
+  id: string;
+  showBackButton: boolean;
+  textResourceBindings: any;
+  triggers?: Triggers[];
+}
+
+export function NavigationBar(props: INavigationBar) {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const orderedLayoutKeys = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.layoutOrder);
+  const returnToView = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.returnToView);
+  const pageTriggers = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.pageTriggers);
+  const triggers = props.triggers || pageTriggers;
+  const textResources = useSelector((state: IRuntimeState) => state.textResources.resources);
+  const currentView: string = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.currentView);
+
+
+
+  const NavBar = () => {
+
+    const [mounted, setMounted] = useState(false);
+
+    useEffect(() => {
+      console.log("Mounted")
+    }, [])
+
+    const OnClickNav = (index: string) => {
+
+      const runPageValidations = !returnToView && triggers?.includes(Triggers.ValidatePage);
+      const runAllValidations = returnToView || triggers?.includes(Triggers.ValidateAllPages);
+      const runValidations = (runAllValidations && 'allPages') || (runPageValidations && 'page') || null;
+
+      dispatch(FormLayoutActions.updateCurrentView({ newView: index, runValidations }));
+
+    };
+
+    const pageList = orderedLayoutKeys.map((view) => 
+      <li className={ currentView == view ? classes.li2 : classes.li } >
+          <a className={ currentView == view ? classes.a2 : classes.a } 
+              onClick={() => OnClickNav(view)}>
+              {getTextResource(view, textResources)}
+          </a>
+      </li>
+    );
+  
+    if(!mounted) {
+      pageList.unshift(
```

Comment:
What is the intent of this?
And what does the mounted state actually do?

Reply:
It's there to make sure this button is only visible when mounted is equal to false
![image](https://user-images.githubusercontent.com/17925860/134633828-a565b442-7ef1-4a5e-89c9-6c2411159a88.png)